Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM jekyll/jekyll as build-env
+
+WORKDIR /src
+
+COPY Gemfile* ./
+RUN chmod 777 /src
+RUN  bundle install
+
+COPY . .
+RUN bundle exec jekyll build --baseurl ''
```

Comment:
The `docker run` process breaks at this point consistently. Here's the error: 

```
Step 7/9 : RUN bundle exec jekyll build --baseurl ''
 ---> Running in c20bcb6aca2a
Configuration file: /src/_config.yml
            Source: /src
       Destination: /src/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
jekyll 3.8.6 | Error:  Permission denied @ rb_sysopen - /src/_site/articles/index.html
The command '/bin/sh -c bundle exec jekyll build --baseurl ''' returned a non-zero code: 1
```

Reply:
this permission issue is supposed to be resolved by:

```Dockerfile
RUN chmod 777 /src
```

I tested it, and strangely does not work for you.  
I have added more permissions, so please test it again after 21391141138a65b9466837aa88ccfae34774bbe5

Reply:
Thanks @ammarnajjar - I did test it and it still didn't work. I was able to get past the errors by checking for the need for the folder and then recursively applying permissions:

```
RUN mkdir -p /src/_site
RUN chmod -R 777 /src/_site
```

That said, the webserver isn't accessible. I'm using `docker-machine` and `docker` client installed through Homebrew. My docker-machine's IP is 192.168.99.100 so I expected to have access to the blog at that IP or on port 4000. What am I missing? 

Thank you! 👍

Reply:
- ok, if `-p` & `-R` work for you, that's great, I will make that change.
- to access your dockerized project, no port is needed, I documented that in the readme, just go to [localhost](http://localhost) 😉

Reply:
run:
```bash
docker build --rm -t mbbroberg:blog . && docker run -it -d -p 80:80 --name blog mbbroberg:blog
```
then open [localhost](http://localhost)

Reply:
so nginx image exposes port `80`, if you want to access your dockerized app on port `4000` you can change the command to:
```bash
docker build --rm -t mbbroberg:blog . && docker run -it -d -p 4000:80 --name blog mbbroberg:blog
```
to map your local `4000` port to nginx `80` port